inject-java/src/main/java/io/micronaut/annotation/processing/visitor/JavaClassElement.java

```diff
@@ -211,37 +211,31 @@ protected void accept(DeclaredType declaringType, javax.lang.model.element.Eleme
                         getterReturnType = mirrorToClassElement(returnType, visitorContext, JavaClassElement.this.genericTypeInfo);
                     }
 
-                    if (getterReturnType != null) {
-
-                        BeanPropertyData beanPropertyData = props.computeIfAbsent(propertyName, BeanPropertyData::new);
-                        configureDeclaringType(declaringTypeElement, beanPropertyData);
-                        beanPropertyData.type = getterReturnType;
-                        beanPropertyData.getter = executableElement;
-                        if (beanPropertyData.setter != null) {
-                            TypeMirror typeMirror = beanPropertyData.setter.getParameters().get(0).asType();
-                            ClassElement setterParameterType = mirrorToClassElement(typeMirror, visitorContext, JavaClassElement.this.genericTypeInfo);
-                            if (setterParameterType == null || !setterParameterType.getName().equals(getterReturnType.getName())) {
-                                beanPropertyData.setter = null; // not a compatible setter
-                            }
+                    BeanPropertyData beanPropertyData = props.computeIfAbsent(propertyName, BeanPropertyData::new);
+                    configureDeclaringType(declaringTypeElement, beanPropertyData);
+                    beanPropertyData.type = getterReturnType;
+                    beanPropertyData.getter = executableElement;
+                    if (beanPropertyData.setter != null) {
+                        TypeMirror typeMirror = beanPropertyData.setter.getParameters().get(0).asType();
+                        ClassElement setterParameterType = mirrorToClassElement(typeMirror, visitorContext, JavaClassElement.this.genericTypeInfo);
+                        if (!setterParameterType.getName().equals(getterReturnType.getName())) {
+                            beanPropertyData.setter = null; // not a compatible setter
                         }
                     }
                 } else if (NameUtils.isSetterName(methodName) && executableElement.getParameters().size() == 1) {
                     String propertyName = NameUtils.getPropertyNameForSetter(methodName);
                     TypeMirror typeMirror = executableElement.getParameters().get(0).asType();
                     ClassElement setterParameterType = mirrorToClassElement(typeMirror, visitorContext, JavaClassElement.this.genericTypeInfo);
 
-                    if (setterParameterType != null) {
```

Comment:
Removing this check doesn't seem correct. The previous call to mirrorToClassElement could have returned null